### ISSUE TITLE
Too many package level constants/variables?

### ISSUE THREAD

**Issue by baba147** [google/gopacket#454]
In my opinion, layers package is too polluted with exported constants/variables. Would it be a good idea to have separate package for each layer?
Perhaps all commonly used layers in a single package (layers), and all uncommon layers in their own packages? (e.g. layers/lldp package).

Any thoughts?

**Comment by vtolstov** [comment]
Does it possible if i'm interest only on ip and dhcp layers iport only needed layers and skip decoding intermediate layers?
For example after ip next is udp, but we don't decode it and skip to udp payload?

**Comment by gconnell** [comment]
No, since you don't know the size of the intermediate layer without decoding it.  Some layers have constant size, but many do not.

**Comment by vtolstov** [comment]
@gconnell but if layer have size in header, why not parse it first and skip the rest decoding?

**Comment by gconnell** [comment]
That actually is currently possible, by overriding the layer decoder.  The short version is that you write your own decoder of UDP that just reads/skips the right size, then you say:

```go
gopacket.OverrideLayerType(LayerTypeUDP, mySkippingUDPDecoder)
```
